### DISCUSSION TITLE
File ../dist/itowns.js   ../dist/debug.js

### DISCUSSION THREAD

**Discussion by Takieddinex** [iTowns/itowns#1832] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/95845810/161570326-c2df1147-40ea-4009-99e8-ed4ddafb75d0.png)
![image](https://user-images.githubusercontent.com/95845810/161571063-74aea7e6-9c73-4253-bc06-0b342df2bf29.png)

Hello ! i'm beginner and i use the example basic 3d tiles to visualiz my own file but i can't find itowns.js and debug.js files.

**Comment by jailln** [comment]
Hi,
If you want to serve it with nginx or apache (which seems to be your case), you have to run `npm install` and `npm run build` in itowns folder to create `itowns.js` and `debug.js` bundles.
Otherwise you can also use npm (which runs a webpack server) by running: `npm install` and `npm start`.

**Reply by Takieddinex** [reply]
Hello, thank you for your reply. However I don't use apache but I'm on windows and as I mentioned before I'm a beginner and I don't really know how it works. I took the HTML code from the 3D Tiles basic example and I tried to add my 3D object by setting up the different Css style folder and JavaScript libraries except that the two libraries deboug and itowns are not loaded.

**Comment by jailln** [comment, marked as answer]
I though you were using Apache since you put itowns folder in `htdocs`. If you're a complete beginner, the simplest way is to start itowns with npm:
* Install node and npm (https://nodejs.org/en/download/)
* Clone or itowns repository
* Open a terminal and change directory to itowns folder
* Run `npm install` to install the dependencies of itowns
* Then run `npm start` : that will launch a local server for itowns, accessible at `http://localhost:8080/`. To access itowns examples: `http://localhost:8080/examples` and `http://localhost:8080/examples/3dtiles_basic.html` for the example that you are interested into.

If you follow these steps, you do not need to css nor js lib filepath to run the examples.

**Reply by Takieddinex** [reply]
![image](https://user-images.githubusercontent.com/95845810/161714649-209ee550-c330-4e0a-9b2b-178561a6084e.png)
Thank you very much, I have tried to do what you have indicated except that I am encountering this error which I cannot resolve

**Comment by jailln** [comment]
Can you list what is in the folder `itowns` please ?

**Reply by Takieddinex** [reply]
![image](https://user-images.githubusercontent.com/95845810/161725534-417e53b5-f89f-45d4-9757-02177d84de18.png)
![image](https://user-images.githubusercontent.com/95845810/161725597-507999d4-160c-4228-8c61-0172c95c3409.png)

it is resolved, thank you, one last question, I would like to view my locally stored file but I can't if you could please to show me how

**Comment by jailln** [comment]
You need to set up a server for your local dataset. For that, either:
* use the [3d-tiles-samples](https://github.com/CesiumGS/3d-tiles-samples) nodejs server. [Follow these instructions](https://github.com/CesiumGS/3d-tiles-samples/blob/main/INSTRUCTIONS.md) for installation and use and put your dataset in the folders served by this server.
* or set up an apache server (you will find a lot of resources on how to do that online)

**Reply by Takieddinex** [reply]
Thank you 👍
